pkg/kubectl/cmd/run.go

```diff
@@ -155,9 +155,9 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 			return err
 		}
 		resourcesList, err := client.Discovery().ServerResources()
+		// ServerResources ignores errors for old servers do not expose discovery
 		if err != nil {
-			// this cover the cases where old servers do not expose discovery
```

Comment:
I agree the error should be exposed here. The discovery client can handle old servers (by ignoring them silently).